src/debugging/errors.md

```diff
@@ -1 +1,58 @@
 # Common Errors
+## Casing Issues
+### Match Nonexhaustive
+Though the code will still compile and run with this warning, it often indicates that the logic in your code does not cover all cases. It is always a good idea to check if you are missing some cases in your thinking, or if the warning comes from a scenario that will never happen according to the function specification. Even if there is a case that may not happen according to the function specification, it may be useful to raise an exception in the "impossible" case to get rid of this warning, because if your code somehow violates the specification and ends up in this case, then you can catch the error and debug it.
+
+### Nested Cases
+When you nest case expressions within case expressions, it's good to wrap your case statements with parentheses. SML will continue to look for patterns to case on, so using parentheses will let it know when to "stop".
+
+For example, the following code will compile without warnings:
+```
+case 0 of
+     0 => (case "x" of _ => 3)
+   | _ => 5
+```
+but this code will have a match redundant error and a match nonexhaustive error:
+```
+case 0 of
+     0 => case "x" of _ => 3
+   | _ => 5
+```
+Obviously, this example is very contrived, but parens may be a source of error if you have nested case expressions.
```

Comment:
parens can help with nested case expressions (I don't think they are the source of error)